fixing the canEdit function

HenrikeW opened the pull request:
I removed the option to pass an individual's id as an argument to the `canEdit` function. Now you can either pass a herd id or a genebank id.

Of those cases where it was used with an individual's id, I removed some entirely where I found them to be unnecessary. In all other cases I pass a herd id instead (sometimes origin herd, sometimes current herd).

HenrikeW committed:
remove herd name from herd validation

not all herds have names and the name is not critical here.

HenrikeW committed:
remove herd name from herd validation

not all herds have names and the name is not critical here.

HenrikeW committed:
move breeding list into own tab

also introduce option to make the popup fullwidth or not.

HenrikeW committed:
move breeding list into own tab

also introduce option to make the popup fullwidth or not.

HenrikeW committed:
add dropdown to choose a herd in individual_form

HenrikeW committed:
add dropdown to choose a herd in individual_form

HenrikeW committed:
pass genebank prop further to individual_form

HenrikeW committed:
pass genebank prop further to individual_form

HenrikeW committed:
reorder individual_form

HenrikeW committed:
reorder individual_form

HenrikeW committed:
autofill number when birth date is chosen

HenrikeW committed:
autofill number when birth date is chosen

jonandernovella committed:
Add method for next ind number

This should create a new valid number for an individual based on breeding events on that year of a given herd.

jonandernovella committed:
Make ind number optional in form

It can be either system generated or passed in the form.

jonandernovella committed:
Update data access tests

They should now reflect the use case when we want the system to generate ind numbers.

airenzp committed:
moved logo and improved individual view

airenzp committed:
moved logo and improved individual view

airenzp committed:
Merge branch 'develop' into fix/validation_issues

airenzp committed:
Merge branch 'fix/layout' into fix/validation_issues

airenzp committed:
Merge branch 'fix/validation_issues' of github.com:NBISweden/theherdbook into fix/validation_issues

HenrikeW committed:
remove recently introduced bug in individual_form

now it even behaves the right way when used for certificates.

HenrikeW committed:
remove recently introduced bug in individual_form

now it even behaves the right way when used for certificates.

jonandernovella committed:
Update app/utils/database.py

Co-authored-by: Henrike Wiemker <65461017+HenrikeW@users.noreply.github.com>

HenrikeW committed:
redo styling of individual_add to start with ancestors

HenrikeW committed:
redo styling of individual_add to start with ancestors

HenrikeW committed:
limit herd options to parents current herds

HenrikeW committed:
limit herd options to parents current herds

HenrikeW committed:
add extra validation for individual number format

HenrikeW committed:
add extra validation for individual number format

HenrikeW committed:
add parents names to breeding event table

HenrikeW committed:
add parents names to breeding event table

HenrikeW committed:
canedit in individual_view based on herds

HenrikeW committed:
canedit in individual_view based on herds

HenrikeW committed:
remove restriction from individual data fetch

HenrikeW committed:
remove restriction from individual data fetch

HenrikeW committed:
access to register page only for admins and managers

HenrikeW committed:
access to register page only for admins and managers

HenrikeW committed:
move canManage and remove canedit from individual_form

HenrikeW committed:
move canManage and remove canedit from individual_form

HenrikeW committed:
remove isNew state from individual_edit

HenrikeW committed:
remove isNew state from individual_edit

HenrikeW committed:
remove canedit from data fetch in individual_edit

HenrikeW committed:
remove canedit from data fetch in individual_edit

HenrikeW committed:
adjust disabled-rules in individual_edit

HenrikeW committed:
adjust disabled-rules in individual_edit

HenrikeW committed:
change fucntion definition of canedit

now it only takes a herd id or genebank id as argument.

HenrikeW committed:
change fucntion definition of canedit

now it only takes a herd id or genebank id as argument.

jonandernovella committed:
Add frontend call for creating inds

jonandernovella committed:
Add better validation in backend

Validate date objects' type and do not set None valued properties.

jonandernovella committed:
Make a few individual properties optional

jonandernovella committed:
Improve frontend error handling

HenrikeW committed:
fix interfaces for Breeding and ExtendedBreeding

HenrikeW committed:
fix interfaces for Breeding and ExtendedBreeding

HenrikeW committed:
keep litter number in resetSibling

HenrikeW committed:
keep litter number in resetSibling

HenrikeW committed:
Merge branch 'fix/validation_issues' of github.com:NBISweden/theherdbook into fix/validation_issues

jonandernovella committed:
Merge pull request #351 from NBISweden/fix/validation_issues

Tweaks on the individual form and herd view

jonandernovella committed:
Fix format error

jonandernovella committed:
Clean up backend additions

jonandernovella committed:
Remove not needed checks

HenrikeW committed:
add function for format date strign

HenrikeW committed:
make things much shorter

HenrikeW committed:
format navigation.tsx with Prettier

HenrikeW committed:
add Swedish theme to Material UI

jonandernovella committed:
Fix generated new ind numbers

The current herd should be checked just in case. Also, we check for the exception when there is no individuals belonging to the breeding event yet.

jonandernovella committed:
Remove unused line

jonandernovella committed:
Remove unused import

pontus committed:
Use persistent storage (mapped directory) for minio, create bucket on start

HenrikeW committed:
Merge pull request #358 from NBISweden/fix/swedish_mui

Swedish text in design elements

jonandernovella committed:
Use origin herd and relocate query scope

jonandernovella committed:
Reformat code

HenrikeW committed:
make links in individual_view open popups

jonandernovella committed:
Merge pull request #357 from NBISweden/bugfix/datestring

fixing date format on certificate preview

pontus reviewed (approved):
Probably ok. Can't trigger any bad case with a short testing session at least.

jonandernovella reviewed (approved):
Looks good. I was confused in the beginning about the "herd" property. I think we could have named it current_herd.

HenrikeW committed:
Merge pull request #350 from NBISweden/dev/new_individuals

Allow generation of new(!) individual entries

jonandernovella committed:
Merge pull request #359 from NBISweden/fix/popup_links

make links in individual_view open popups

jonandernovella committed:
Merge pull request #361 from NBISweden/feature/s3_develop_bucket

Use persistent storage (mapped directory) for minio, create bucket at start

jonandernovella committed:
Do execute the kull count query

Currently we were generating ids based on the length of the SQL query string...

jonandernovella committed:
Merge pull request #364 from NBISweden/hotfix/sql-exec

Do execute the litter count query

HenrikeW committed:
clean rebase mess

HenrikeW committed:
Merge branch 'bugfix/canedit' of github.com:NBISweden/theherdbook into bugfix/canedit
